services/consensusalgo/leanhelixconsensus/network_communication.go

```diff
@@ -0,0 +1,64 @@
+package leanhelixconsensus
+
+import (
+	"context"
+	"github.com/orbs-network/lean-helix-go"
+	lhprimitives "github.com/orbs-network/lean-helix-go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/consensus"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
+	"github.com/orbs-network/orbs-spec/types/go/services/gossiptopics"
+)
+
+func (s *service) RequestOrderedCommittee(seed uint64) []lhprimitives.Ed25519PublicKey {
+	panic("implement me")
+}
+
+func (s *service) IsMember(pk lhprimitives.Ed25519PublicKey) bool {
+	panic("implement me")
+}
+
+type leanHelixHandlerImpl struct {
+	handlerFunc func(ctx context.Context, message leanhelix.ConsensusRawMessage)
+}
+
+// Lib calls this method to register itself for incoming messages, and supplies the callback
+func (s *service) RegisterOnMessage(onReceivedMessage func(ctx context.Context, message leanhelix.ConsensusRawMessage)) int {
```

Comment:
probably better to pass an interface than a callback method

Reply:
It would only slightly reduce complexity as it just hides the need to sync the function signature between lib and Orbs but does not prevent it. Also the obvious name "MessageReceiver" is already used by the lib